### DISCUSSION TITLE
Can I give read-only access to a private repo (from a developer account)?

### DISCUSSION THREAD

**Discussion by visakhkumar** [community/community#23547] [New to GitHub]
<p>Can I give read-only access to my private repo, to someone who is not a collaborator at present? Can I share a url, or add someone as a read-only collaborator? I don’t want to give fork/clone/download rights.</p>

**Comment by Gisgar3** [comment, marked as answer]
<p>Based on the official <strong>GitHub Help</strong>  <a href="https://help.github.com/en/articles/about-repositories" rel="nofollow noopener">article</a> regarding user-repositories, only collaborators and repository owners are capable of viewing private repositories.</p>
<p>Hope this helped!</p>

**Comment by johnochinero** [comment]
<p>Thanks! The question stands, are there different levels of collaboration to block any contamination of my files? “Read” only collaboration, Mirror my account for safe keeping or whatever.</p>
<p>Or if I invite an external developer to access my account my entire life is placed in jeopardy?</p>
<p>Thanks,</p>
<p>John</p>

**Comment by johnochinero** [comment]
<p>Thank you again for your response.</p>
<p>My repositories are “Public”  with no collaborators.</p>
<p>Shouldn’t Mobile App Development companies be able to view the details of the repositories and gather sufficient information to prepare a proposal on such things as “Deployment to AWS” or “clone mobile app for White Box Sales”?</p>
<p>Thanks again.</p>
<p>John</p>

**Comment by johnochinero** [comment]
<p>This seems odd. So, if a 3rd party needs access to my codes/respository to evaluate and make a proposal for potential project, then I need to give them full rights to be able to destroy my code and I lose my business?</p>
<p>Is there an alternative, mirror my codes so I have something for safe keeping. I find it hard to believe that my codes are placed in jeopardy for a simple evaluation by a 3rd party.</p>
<p>Please clarify.</p>
<p>Thanks!</p>

**Comment by venkataprasannam** [comment]
Hi team,

Can I restrict a collaborator from downloading or cloning an archived repository?

Please let me know. Thank you for your support!
